scripts/pre_e2e.sh

```diff
@@ -4,5 +4,18 @@ set -Eeuo pipefail
 # Generate big files which are used during testing
 # `uploadProject.spec.js` -> should reject files bigger than `MAX_FILE_SIZE`
 auto_gen_fixtures_path=e2e/cypress/fixtures/auto-gen
-mkdir $auto_gen_fixtures_path
-truncate -s "${MAX_FILE_SIZE}" "${auto_gen_fixtures_path}/big.txt"
+rm -rf $auto_gen_fixtures_path && mkdir $auto_gen_fixtures_path
+truncate -s "${MAX_FILE_SIZE:-6M}" "${auto_gen_fixtures_path}/big.txt"
+
+# generate admin token used for importing projects during e2e tests
+if [[ "${CI:-}" = "true" ]]; then
+    docker-compose -f docker-compose.yml -f docker-compose.deploy.yml up -d
+    until docker logs kitspace_gitea_1 | grep -q 'ORM engine initialization successful' ; do sleep 3s; done
+else
+    # The tests are running locally.
+    docker-compose up -d
+fi
+
+token="$(deno run --allow-env --allow-net --allow-run ./scripts/importBoardsTxt.ts --tokenOnly)"
+# reutrn the token so it can be used as $(./scripts/pre_e2e.sh)
+echo $token
```

Comment:
I don't think giving this script two purposes is a good idea.

Reply:
I think of it as the stuff we need to do before running e2e (the build step), and like any other build step, it can involve multiple steps.

Reply:
I guess but if the instructions say `yarn --cwd e2e gui --env GITEA_ADMIN_TOKEN=$(scripts/pre_e2e.sh)` then the fact that it also creates a test-data file when you do that is surprising behaviour. I suggest just saying:

```
GITEA_ADMIN_TOKEN="$(deno run --allow-env --allow-net --allow-run ./scripts/importBoardsTxt.ts --tokenOnly)"
yarn --cwd e2e gui --env GITEA_ADMIN_TOKEN="${GITEA_ADMIN_TOKEN}"
```
(shouldn't it by `CYPRESS_GITEA_ADMIN_TOKEN` anyway?)

Ideally `pre_e2e.sh` should be called `generate_test_fixture.sh` or similar so it's less tempting to make it a dumping ground. Also `importBoardsTxt.ts --tokenonly` should be a separate script (`createGiteaAdminToken.ts`) but I'm willing to let these go in the interest of shipping. 

In general you should think twice if you are making any script/function/class/whatever do two things because it makes what it's going to do much less clear.

Reply:
Resolved in 0872bab.